### PR TITLE
[ui] convert project keys to react

### PR DESCRIPTION
**Pull request by dcramer**
#nochanges

### CODE REVIEW

File: src/sentry/static/sentry/app/views/projectKeys.jsx [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+          loading: false
+        });
+      }
+    });
+  },
+
+  handleRemoveKey(data) {
+    this.setState(state => {
+      return {
+        keyList: state.keyList.filter(key => {
```
Comment (benvinegar) [inline src/sentry/static/sentry/app/views/projectKeys.jsx]: FYI you can do this as a one-liner (hmm, does Prettier convert this?):

```javascript
keyList: state.keyList.filter(key => key.id !== data.id)
```

**Review by benvinegar — COMMENTED** [review]
Seems like a good opportunity to use Jest's snapshot testing.

I went through the code twice, nothing really stuck out except a typo (see comments). Will do some quick manual QA.

File: src/sentry/static/sentry/app/views/projectKeys.jsx (hunk 1 above)
Comment (dcramer) [inline src/sentry/static/sentry/app/views/projectKeys.jsx]: i do it as two lines because its easier to read often

**Comment by dcramer** [comment]
Revised a few things:

- Cleaned up language
- Add titles (keep forgetting about this)
- Removed legacy tests/fixed some minor breakage

**Pull request merged**